Update to glutin v0.21.0

Glutin v0.21.0 migration guide: https://gentz.rocks/posts/glutin-v0-21-0-migration-guide/

Currently using a minor fork of glutin, https://github.com/iceiix/glutin/pull/1 
Updating to 0.21.0, errors:

```
error[E0432]: unresolved import `glutin::ContextTrait`
  --> src/gl/mod.rs:16:5
   |
16 | use glutin::ContextTrait;
   |     ^^^^^^^^^^^^^^^^^^^^ no `ContextTrait` in the root. Did you mean to use `ContextError`?

error[E0432]: unresolved import `glutin::ContextTrait`
  --> src/main.rs:55:5
   |
55 | use glutin::ContextTrait;
   |     ^^^^^^^^^^^^^^^^^^^^ no `ContextTrait` in the root. Did you mean to use `ContextError`?

warning: unused import: `glutin::ContextTrait`
  --> src/gl/mod.rs:16:5
   |
16 | use glutin::ContextTrait;
   |     ^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(unused_imports)] on by default

warning: unused import: `glutin::ContextTrait`
  --> src/main.rs:55:5
   |
55 | use glutin::ContextTrait;
   |     ^^^^^^^^^^^^^^^^^^^^

error[E0107]: wrong number of type arguments: expected 1, found 0
   --> src/main.rs:358:37
    |
358 | fn handle_window_event(window: &mut glutin::WindowedContext,
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^ expected 1 type argument

error[E0107]: wrong number of type arguments: expected 1, found 0
  --> src/gl/mod.rs:25:20
   |
25 | pub fn init(vid: & glutin::WindowedContext) {
   |                    ^^^^^^^^^^^^^^^^^^^^^^^ expected 1 type argument

error: aborting due to 4 previous errors

Some errors occurred: E0107, E0432.
For more information about an error, try `rustc --explain E0107`.
error: Could not compile `stevenarella`.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,7 +21,8 @@ opt-level = 1
 cfg-if = "0.1.9"
 wasm-bindgen = "0.2.44"
 sha-1 = "0.8.1"
-glutin = "0.21.0"
+#glutin = { git = "https://github.com/iceiix/glutin", branch = "wasm_stub" }
+glutin = { git = "https://github.com/iceiix/glutin", rev = "5f490a9ffec1e1facbdad2264585301e8a5927cd" }
 byteorder = "1.2.7"
 serde = "1.0.91"
 serde_json = "1.0.39"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,7 +22,7 @@ cfg-if = "0.1.9"
 wasm-bindgen = "0.2.44"
 sha-1 = "0.8.1"
 #glutin = { git = "https://github.com/iceiix/glutin", branch = "wasm_stub" }
-glutin = { git = "https://github.com/iceiix/glutin", rev = "5f490a9ffec1e1facbdad2264585301e8a5927cd" }
+glutin = { git = "https://github.com/iceiix/glutin", rev = "43c166146f6999afc99a2159c872b1dabd2e877f" }
 byteorder = "1.2.7"
 serde = "1.0.91"
 serde_json = "1.0.39"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,7 +21,6 @@ opt-level = 1
 cfg-if = "0.1.9"
 wasm-bindgen = "0.2.44"
 sha-1 = "0.8.1"
-#glutin = { git = "https://github.com/iceiix/glutin", branch = "wasm_stub" }
 glutin = { git = "https://github.com/iceiix/glutin", rev = "5ff18dec039f6854091685980d7213858a8aaa4c" }
 byteorder = "1.2.7"
 serde = "1.0.91"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,7 +22,7 @@ cfg-if = "0.1.9"
 wasm-bindgen = "0.2.44"
 sha-1 = "0.8.1"
 #glutin = { git = "https://github.com/iceiix/glutin", branch = "wasm_stub" }
-glutin = { git = "https://github.com/iceiix/glutin", rev = "43c166146f6999afc99a2159c872b1dabd2e877f" }
+glutin = { git = "https://github.com/iceiix/glutin", rev = "5ff18dec039f6854091685980d7213858a8aaa4c" }
 byteorder = "1.2.7"
 serde = "1.0.91"
 serde_json = "1.0.39"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,7 +21,7 @@ opt-level = 1
 cfg-if = "0.1.9"
 wasm-bindgen = "0.2.44"
 sha-1 = "0.8.1"
-glutin = { git = "https://github.com/iceiix/glutin", rev = "de52fba20678f3931fba3eb6700774469e48512f" }
+glutin = "0.21.0"
 byteorder = "1.2.7"
 serde = "1.0.91"
 serde_json = "1.0.39"

--- a/src/gl/mod.rs
+++ b/src/gl/mod.rs
@@ -13,7 +13,6 @@
 // limitations under the License.
 
 extern crate steven_gl as gl;
-use glutin::ContextTrait;
 
 use std::ops::BitOr;
 use std::ffi;

--- a/src/gl/mod.rs
+++ b/src/gl/mod.rs
@@ -21,7 +21,7 @@ use std::ptr;
 use std::ops::{Deref, DerefMut};
 
 /// Inits the gl library. This should be called once a context is ready.
-pub fn init(vid: & glutin::WindowedContext) {
+pub fn init<T>(vid: & glutin::WindowedContext<T>) where T: glutin::ContextCurrentState {
     gl::load_with(|s| vid.get_proc_address(s) as *const _);
 }
 

--- a/src/gl/mod.rs
+++ b/src/gl/mod.rs
@@ -21,7 +21,7 @@ use std::ptr;
 use std::ops::{Deref, DerefMut};
 
 /// Inits the gl library. This should be called once a context is ready.
-pub fn init<T>(vid: & glutin::WindowedContext<T>) where T: glutin::ContextCurrentState {
+pub fn init(vid: & glutin::WindowedContext<glutin::PossiblyCurrent>) {
     gl::load_with(|s| vid.get_proc_address(s) as *const _);
 }
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -52,7 +52,6 @@ use std::thread;
 use std::sync::mpsc;
 use crate::protocol::mojang;
 use glutin;
-use glutin::ContextTrait;
 
 const CL_BRAND: console::CVar<String> = console::CVar {
     ty: PhantomData,

--- a/src/main.rs
+++ b/src/main.rs
@@ -258,7 +258,7 @@ pub fn main() {
     }
 
     let textures = renderer.get_textures();
-    let dpi_factor = window.get_current_monitor().get_hidpi_factor();
+    let dpi_factor = window.window().get_current_monitor().get_hidpi_factor();
     let default_protocol_version = protocol::versions::protocol_name_to_protocol_version(
         opt.default_protocol_version.unwrap_or("".to_string()));
     let mut game = Game {
@@ -297,8 +297,8 @@ pub fn main() {
         let diff = now.duration_since(last_frame);
         last_frame = now;
         let delta = (diff.subsec_nanos() as f64) / frame_time;
-        let (width, height) = window.get_inner_size().unwrap().into();
-        let (physical_width, physical_height) = window.get_inner_size().unwrap().to_physical(game.dpi_factor).into();
+        let (width, height) = window.window().get_inner_size().unwrap().into();
+        let (physical_width, physical_height) = window.window().get_inner_size().unwrap().to_physical(game.dpi_factor).into();
 
         let version = {
             let try_res = game.resource_manager.try_write();
@@ -381,8 +381,8 @@ fn handle_window_event<T>(window: &mut glutin::WindowedContext<T>,
                 use std::f64::consts::PI;
 
                 if game.focused {
-                    window.grab_cursor(true).unwrap();
-                    window.hide_cursor(true);
+                    window.window().grab_cursor(true).unwrap();
+                    window.window().hide_cursor(true);
                     if let Some(player) = game.server.player {
                         let rotation = game.server.entities.get_component_mut(player, game.server.rotation).unwrap();
                         rotation.yaw -= rx;
@@ -395,8 +395,8 @@ fn handle_window_event<T>(window: &mut glutin::WindowedContext<T>,
                         }
                     }
                 } else {
-                    window.grab_cursor(false).unwrap();
-                    window.hide_cursor(false);
+                    window.window().grab_cursor(false).unwrap();
+                    window.window().hide_cursor(false);
                 }
             },
 
@@ -406,8 +406,8 @@ fn handle_window_event<T>(window: &mut glutin::WindowedContext<T>,
         Event::WindowEvent{event, ..} => match event {
             WindowEvent::CloseRequested => game.should_close = true,
             WindowEvent::Resized(logical_size) => {
-                game.dpi_factor = window.get_hidpi_factor();
-                window.resize(logical_size.to_physical(game.dpi_factor));
+                game.dpi_factor = window.window().get_hidpi_factor();
+                window.window().resize(logical_size.to_physical(game.dpi_factor));
             },
 
             WindowEvent::ReceivedCharacter(codepoint) => {
@@ -419,17 +419,17 @@ fn handle_window_event<T>(window: &mut glutin::WindowedContext<T>,
             WindowEvent::MouseInput{device_id: _, state, button, modifiers: _} => {
                 match (state, button) {
                     (ElementState::Released, MouseButton::Left) => {
-                        let (width, height) = window.get_inner_size().unwrap().into();
+                        let (width, height) = window.window().get_inner_size().unwrap().into();
 
                         if game.server.is_connected() && !game.focused && !game.screen_sys.is_current_closable() {
                             game.focused = true;
-                            window.grab_cursor(true).unwrap();
-                            window.hide_cursor(true);
+                            window.window().grab_cursor(true).unwrap();
+                            window.window().hide_cursor(true);
                             return;
                         }
                         if !game.focused {
-                            window.grab_cursor(false).unwrap();
-                            window.hide_cursor(false);
+                            window.window().grab_cursor(false).unwrap();
+                            window.window().hide_cursor(false);
                             ui_container.click_at(game, game.last_mouse_x, game.last_mouse_y, width, height);
                         }
                     },
@@ -447,7 +447,7 @@ fn handle_window_event<T>(window: &mut glutin::WindowedContext<T>,
                 game.last_mouse_y = y;
 
                 if !game.focused {
-                    let (width, height) = window.get_inner_size().unwrap().into();
+                    let (width, height) = window.window().get_inner_size().unwrap().into();
                     ui_container.hover_at(game, x, y, width, height);
                 }
             },
@@ -467,13 +467,13 @@ fn handle_window_event<T>(window: &mut glutin::WindowedContext<T>,
                 match (input.state, input.virtual_keycode) {
                     (ElementState::Released, Some(VirtualKeyCode::Escape)) => {
                         if game.focused {
-                            window.grab_cursor(false).unwrap();
-                            window.hide_cursor(false);
+                            window.window().grab_cursor(false).unwrap();
+                            window.window().hide_cursor(false);
                             game.focused = false;
                             game.screen_sys.replace_screen(Box::new(screen::SettingsMenu::new(game.vars.clone(), true)));
                         } else if game.screen_sys.is_current_closable() {
-                            window.grab_cursor(true).unwrap();
-                            window.hide_cursor(true);
+                            window.window().grab_cursor(true).unwrap();
+                            window.window().hide_cursor(true);
                             game.focused = true;
                             game.screen_sys.pop_screen();
                         }
@@ -483,9 +483,9 @@ fn handle_window_event<T>(window: &mut glutin::WindowedContext<T>,
                     },
                     (ElementState::Pressed, Some(VirtualKeyCode::F11)) => {
                         if !game.is_fullscreen {
-                            window.set_fullscreen(Some(window.get_current_monitor()));
+                            window.window().set_fullscreen(Some(window.window().get_current_monitor()));
                         } else {
-                            window.set_fullscreen(None);
+                            window.window().set_fullscreen(None);
                         }
 
                         game.is_fullscreen = !game.is_fullscreen;

--- a/src/main.rs
+++ b/src/main.rs
@@ -407,7 +407,7 @@ fn handle_window_event(window: &mut glutin::WindowedContext<glutin::PossiblyCurr
             WindowEvent::CloseRequested => game.should_close = true,
             WindowEvent::Resized(logical_size) => {
                 game.dpi_factor = window.window().get_hidpi_factor();
-                window.window().resize(logical_size.to_physical(game.dpi_factor));
+                window.resize(logical_size.to_physical(game.dpi_factor));
             },
 
             WindowEvent::ReceivedCharacter(codepoint) => {

--- a/src/main.rs
+++ b/src/main.rs
@@ -223,13 +223,13 @@ pub fn main() {
     let window_builder = glutin::WindowBuilder::new()
         .with_title("Stevenarella")
         .with_dimensions(glutin::dpi::LogicalSize::new(854.0, 480.0));
-    let context = glutin::ContextBuilder::new()
+    let mut window = glutin::ContextBuilder::new()
         .with_stencil_buffer(0)
         .with_depth_buffer(24)
         .with_gl(glutin::GlRequest::GlThenGles{opengl_version: (3, 2), opengles_version: (2, 0)})
         .with_gl_profile(glutin::GlProfile::Core)
-        .with_vsync(vsync);
-    let mut window = glutin::WindowedContext::new(window_builder, context, &events_loop)
+        .with_vsync(vsync)
+        .build_windowed(window_builder, &events_loop)
         .expect("Could not create glutin window.");
 
     unsafe {

--- a/src/main.rs
+++ b/src/main.rs
@@ -223,7 +223,7 @@ pub fn main() {
     let window_builder = glutin::WindowBuilder::new()
         .with_title("Stevenarella")
         .with_dimensions(glutin::dpi::LogicalSize::new(854.0, 480.0));
-    let mut window = glutin::ContextBuilder::new()
+    let window = glutin::ContextBuilder::new()
         .with_stencil_buffer(0)
         .with_depth_buffer(24)
         .with_gl(glutin::GlRequest::GlThenGles{opengl_version: (3, 2), opengles_version: (2, 0)})
@@ -232,9 +232,9 @@ pub fn main() {
         .build_windowed(window_builder, &events_loop)
         .expect("Could not create glutin window.");
 
-    unsafe {
-        window.make_current().expect("Could not set current context.");
-    }
+    let window = unsafe {
+        window.make_current().expect("Could not set current context.")
+    };
 
     gl::init(&window);
 
@@ -354,10 +354,10 @@ pub fn main() {
     }
 }
 
-fn handle_window_event<T>(window: &mut glutin::WindowedContext<T>,
+fn handle_window_event(window: &mut glutin::WindowedContext<glutin::PossiblyCurrent>,
                        game: &mut Game,
                        ui_container: &mut ui::Container,
-                       event: glutin::Event) where T: glutin::ContextCurrentState {
+                       event: glutin::Event) {
     use glutin::*;
     match event {
         Event::DeviceEvent{event, ..} => match event {

--- a/src/main.rs
+++ b/src/main.rs
@@ -354,10 +354,10 @@ pub fn main() {
     }
 }
 
-fn handle_window_event(window: &mut glutin::WindowedContext,
+fn handle_window_event<T>(window: &mut glutin::WindowedContext<T>,
                        game: &mut Game,
                        ui_container: &mut ui::Container,
-                       event: glutin::Event) {
+                       event: glutin::Event) where T: glutin::ContextCurrentState {
     use glutin::*;
     match event {
         Event::DeviceEvent{event, ..} => match event {

--- a/src/main.rs
+++ b/src/main.rs
@@ -232,7 +232,7 @@ pub fn main() {
         .build_windowed(window_builder, &events_loop)
         .expect("Could not create glutin window.");
 
-    let window = unsafe {
+    let mut window = unsafe {
         window.make_current().expect("Could not set current context.")
     };
 

